### PR TITLE
CASSANDRA-18511 Add support for JMX in the in-jvm dtest framework (4.0)

### PR DESCRIPTION
**Pull request by JeetKunDoug**

### CODE REVIEW

File: relocate-dependencies.pom [hunk 1]
```diff
@@ -67,7 +67,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>3.2.1</version>
+                <version>3.4.1</version>
```
Comment (jonmeredith) [inline relocate-dependencies.pom:70]: is this bump required to make things work?

File: test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java [hunk 2]
```diff
@@ -70,11 +70,11 @@
 import org.apache.cassandra.distributed.api.NodeToolResult;
 import org.apache.cassandra.distributed.api.TokenSupplier;
 import org.apache.cassandra.distributed.shared.InstanceClassLoader;
-import org.apache.cassandra.distributed.shared.Isolated;
+import org.apache.cassandra.utils.Isolated;
 import org.apache.cassandra.distributed.shared.MessageFilters;
 import org.apache.cassandra.distributed.shared.Metrics;
 import org.apache.cassandra.distributed.shared.NetworkTopology;
-import org.apache.cassandra.distributed.shared.Shared;
+import org.apache.cassandra.utils.Shared;
```
Comment (jonmeredith) [inline test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java]: should re-order position in import list.

File: test/distributed/org/apache/cassandra/distributed/impl/Instance.java [hunk 3]
```diff
@@ -815,6 +1012,9 @@ public Future<Void> shutdown(boolean graceful)
             
             // ScheduledExecutors shuts down after MessagingService, as MessagingService may issue tasks to it.
             error = parallelRun(error, executor, () -> ScheduledExecutors.shutdownAndWait(1L, MINUTES));
+            
+            error = parallelRun(error, executor,
+                                this::stopJmx);
```
Comment (jonmeredith) [inline test/distributed/org/apache/cassandra/distributed/impl/Instance.java]: single line like 3.11?

File: test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java [hunk 4]
```diff
@@ -35,7 +35,7 @@
 import org.apache.cassandra.distributed.api.Feature;
 import org.apache.cassandra.distributed.api.IInstanceConfig;
 import org.apache.cassandra.distributed.shared.NetworkTopology;
-import org.apache.cassandra.distributed.shared.Shared;
+import org.apache.cassandra.utils.Shared;
 import org.apache.cassandra.distributed.upgrade.UpgradeTestBase;
```
Comment (jonmeredith) [inline test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java]: fix import ordering

**Review by jonmeredith — APPROVED** [review]

File: relocate-dependencies.pom (hunk 1 above)
Comment (JeetKunDoug) [reply relocate-dependencies.pom:70]: Yup - it's a back port from 4.1 to make the shaded dtest jar build correctly.

File: test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java (hunk 2 above)
Comment (JeetKunDoug) [reply test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java]: Done

File: test/distributed/org/apache/cassandra/distributed/impl/Instance.java (hunk 3 above)
Comment (JeetKunDoug) [reply test/distributed/org/apache/cassandra/distributed/impl/Instance.java]: Done

File: test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java (hunk 4 above)
Comment (JeetKunDoug) [reply test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java]: Done

**Comment by JeetKunDoug** [comment]
Thanks for the review(s) @jonmeredith - greatly appreciated. Should be all set now.

File: test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java [hunk 5]
```diff
@@ -70,13 +70,13 @@
...
 import org.apache.cassandra.distributed.shared.InstanceClassLoader;
-import org.apache.cassandra.distributed.shared.Isolated;
 import org.apache.cassandra.distributed.shared.MessageFilters;
 import org.apache.cassandra.distributed.shared.Metrics;
 import org.apache.cassandra.distributed.shared.NetworkTopology;
-import org.apache.cassandra.distributed.shared.Shared;
 import org.apache.cassandra.distributed.shared.ShutdownException;
 import org.apache.cassandra.distributed.shared.Versions;
+import org.apache.cassandra.utils.Isolated;
+import org.apache.cassandra.utils.Shared;
```
Comment (jonmeredith) [inline test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java]: not sure if the UI isn't updating right, but theses two should be after `org.apache.cassandra.utils.FBUtilities`

File: test/distributed/org/apache/cassandra/distributed/impl/Instance.java [hunk 6]
```diff
@@ -88,6 +101,7 @@
 import org.apache.cassandra.distributed.mock.nodetool.InternalNodeProbeFactory;
 import org.apache.cassandra.distributed.shared.Metrics;
 import org.apache.cassandra.gms.ApplicationState;
+import org.apache.cassandra.utils.ReflectionUtils;
```
Comment (jonmeredith) [inline test/distributed/org/apache/cassandra/distributed/impl/Instance.java]: import should be later in file

**Review by jonmeredith — APPROVED** [review]

**Pull request closed without merging**